anxcloud/provider.go

```diff
@@ -76,7 +82,20 @@ func providerConfigure(ctx context.Context, d *schema.ResourceData) (interface{}
 		return nil, diags
 	}
 
-	return c, diags
+	apiClient, err := api.NewAPI(api.WithClientOptions(opts...))
+	if err != nil {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "Unable to create generic Anexia API",
```

Comment:
phrasing, "Unable to create Anexia generic API client" - let's always call it "generic client", adding "API" when not clear where the message is shown :)